server.js

```diff
@@ -38,9 +38,10 @@ module.exports = !WebSocket.Server ? null : function (opts, onConnection) {
     proxy(server, 'request')
     proxy(server, 'close')
 
-    wsServer.on('connection', function (socket) {
+    wsServer.on('connection', function (socket, req) {
+      socket.upgradeReq = req // mix: kinda gross hack to preserve the API of duplex.js, but might confuse users...
```

Comment:
ok this line gets the tests passing for : 
- this module
- `ssb-server/test/bin.js`

This fix would stop us needing a breaking change ... but feels a bit ech? don't know how much duplex.js is getting used directly

Reply:
I think a cleaner option would be to pass the headers directly to `duplex` (I don't think anything uses it directly, but client.js and server.js need it) or, at least, to pass just the origin header. A WebSocket object created in the browser doesn't even have a headers property, but multiserver sets the address from the address you passed it, in that case.

Reply:
oh I also ran this against the multiserver tests and found another place it breaks.
however, in this case, I think it's multiserver that is causing the problem...

Reply:
So you mean we should change the code to something like  : 

```js
      var opts = {
        header: req.headers,
        remoteAddress: req.connection.remoteAddress
      }
      var stream = ws(socket, opts)   // here ws is duplex.js
      emitter.emit('connection', stream)
```